test/SwapToRatioTest.spec.ts

```diff
@@ -0,0 +1,662 @@
+          tickUpper: TICK_SPACINGS[FeeAmount.MEDIUM] * 1,
+          amount0Desired: expandTo18Decimals(1_000),
+          amount1Desired: expandTo18Decimals(1_000),
+          amount0Min: 0,
+          amount1Min: 0,
+          recipient: wallets[0].address,
+          deadline: 1,
+        })
+
+        liquidity = await pool.liquidity()
+        sqrtRatioX96 = (await pool.slot0()).sqrtPriceX96
+      })
+
+      describe('initial deposit has excess of token0', async () => {
+        it('returns token amounts which reflect a real post swap amounts', async () => {
+          amount0Initial = expandTo18Decimals(30_000)
+          amount1Initial = expandTo18Decimals(1_000)
+
+          const { doSwap, amount0Updated, amount1Updated } = await swapToNextInitializedTick(swapToRatio, {
+            amount0Initial,
+            amount1Initial,
+            sqrtRatioX96Lower,
+            sqrtRatioX96Upper,
+            sqrtRatioX96,
+            liquidity,
+            fee,
+            sqrtRatioX96Target,
+            zeroForOne: true,
+          })
+
+          const amountToSwap = await swapToRatio.getAmount0Delta(sqrtRatioX96, sqrtRatioX96Target, liquidity, false)
+          await tokens[0].approve(router.address, amountToSwap.abs())
+          const amountOut = await router.callStatic.exactInputSingle({
+            tokenIn: token0.address,
+            tokenOut: token1.address,
+            fee: FeeAmount.MEDIUM,
+            recipient: wallets[0].address,
+            deadline: 1,
+            amountIn: amountToSwap.abs(),
+            amountOutMinimum: 0,
+            sqrtPriceLimitX96: expandTo18Decimals(100_000),
+          })
+
+          expect(amountOut.add(amount1Initial)).to.equal(amount1Updated)
+        })
+
+        it('moves the pool price to the targetPrice', async () => {
+          amount0Initial = expandTo18Decimals(30_000)
+          amount1Initial = expandTo18Decimals(1_000)
+
+          const { doSwap, amount0Updated, amount1Updated } = await swapToNextInitializedTick(swapToRatio, {
+            amount0Initial,
+            amount1Initial,
+            sqrtRatioX96Lower,
+            sqrtRatioX96Upper,
+            sqrtRatioX96,
+            liquidity,
+            fee,
+            sqrtRatioX96Target,
+            zeroForOne: true,
+          })
+
+          const amountToSwap = await swapToRatio.getAmount0Delta(sqrtRatioX96, sqrtRatioX96Target, liquidity, false)
+          await tokens[0].approve(router.address, amountToSwap.abs())
+          await router.exactInputSingle({
+            tokenIn: token0.address,
+            tokenOut: token1.address,
+            fee: FeeAmount.MEDIUM,
+            recipient: wallets[0].address,
+            deadline: 1,
+            amountIn: amountToSwap.abs(),
+            amountOutMinimum: 0,
+            sqrtPriceLimitX96: expandTo18Decimals(100_000),
+          })
+
+          expect((await pool.slot0()).sqrtPriceX96).to.eq(sqrtRatioX96)
```

Comment:
```
AssertionError: Expected "78991555868274566570269357078" to be equal 79228162514264337593543950336
```

would work backwards from swapAmounts first to see if this improves in the process